lib/store/settings_store.dart

```diff
@@ -1708,16 +1708,40 @@ abstract class SettingsStoreBase with Store {
     powNodes[walletType] = node;
   }
 
-  void initializeTrocadorProviderStates() {
-    for (var provider in TrocadorExchangeProvider.availableProviders) {
-      final savedState = _sharedPreferences.getBool(provider) ?? true;
-      trocadorProviderStates[provider] = savedState;
+  @action
+  Future<void> updateAllTrocadorProviderStates(List<String> availableProviders) async {
+    String? serializedData = await _sharedPreferences.getString(PreferencesKey.trocadorProviderStatesKey);
+    Map<String, bool> regularMap = {};
```

Comment:
for backward compatability, re-add the availableProviders list that we had, and loop through them like it used to be, and fetch the saved preference of the user for this provider and add it to the regular map.
so that users who already have some providers disabled, would still have them disabled with the new flow

and please test it by switching to main, disabling some providers from the hardcoded list we have, and then switch to this branch, and see if they will still be disabled, along side the new providers that would get added and should be enabled by default

Reply:
Added backward compatibility for provider states by restoring saved user preferences for existing providers. Tested by switching from main to ensure previous user settings persist.